pkgs/applications/networking/msmtp/default.nix

```diff
@@ -1,5 +1,6 @@
 { stdenv, lib, fetchurl, autoreconfHook, pkgconfig
 , openssl, netcat-gnu, gnutls, gsasl, libidn, Security
+, libsecret, withKeyring ? true
```

Comment:
I believe the canonical way to do this is:
```, withKeyring ? true, libsecret ? null```

Reply:
why do it only for libsecret ? I don't see `libidn ? null, gnutls ? null`

Reply:
The difference I see is that those are already required dependencies, but as you are proposing `libsecret` as an optional dependency.

My way of doing it would be without the `withKeyring` parameter and use the `isNull` builtin for the configure flags (nothing would be needed for `buildInputs` IIRC).

Reply:
I see the point of not having to mirror every configure flag into the nix file.
I removed the configureFlag ("--with-libsecret" is enabled by default) but if I remove libsecret from buildinputs, it won't be found by the configure (confirmed by trial).

Reply:
The ```withFoo``` pattern as opposed to passing ```libsecret = null;``` and then use ```isNull``` is to check if something should be enabled or disabled is that often multiple derivations are added to ```buildInputs``` as a consequence of toggling a single flag. So using ```withFoo``` makes it easier to read and easier to adjust. When it's only a single derivation, it really doesn't make much of a difference, but for the sake of uniformity I personally prefer it that way.

Reply:
not sure I got your last sentence: so you would prefer with withKeyring ? and the libsecret ? null is just to hint at thepackage being optional ?
which maintainer should I listen to :D ?
I prefer the withKeyring approach (explicit over implicit)

Reply:
> you would prefer with withKeyring ? and the libsecret ?

```, withKeyring ? true, libsecret ? null```

and then:
```
  buildInputs = [ openssl gnutls gsasl libidn ]
    ++ stdenv.lib.optional withKeyring     libsecret
    ++ stdenv.lib.optional stdenv.isDarwin Security;
```

> which maintainer should I listen to :D ?

Me of course! Why would you even ask that?!?! 😃